users/d4mation/macros.c

```diff
@@ -0,0 +1,224 @@
+#include "d4mation.h"
+#include "tap-hold.h"
+#include "zalgo.h"
+#include "macros.h"
+
+bool zalgo_enabled = false;
+
+bool process_record_user( uint16_t keycode, keyrecord_t *record ) {
+
+  switch ( keycode ) {
+
+    case _GRAVE_ESC:
+
+      /* Send ` on Tap, Esc on Hold */
+      tap_or_hold( record, KC_GRAVE, KC_ESC );
+
+      return false;
+      break;
+
+    case PHPOPEN:
+
+      if ( record->event.pressed ) {
+
+        tap_code16( S( KC_COMMA ) );
+        tap_code16( S( KC_SLASH ) );
+
+        tap_code( KC_P );
+        tap_code( KC_H );
+        tap_code( KC_P );
+
+      }
+
+      return false;
+      break;
+
+    case PHPCLSE:
+
+      if ( record->event.pressed ) {
+        tap_code16( S( KC_SLASH ) );
+        tap_code16( S( KC_DOT ) );
+      }
+
+      return false;
+      break;
+
+    case AMENO:  /* ༼ つ ◕_◕ ༽つ */
+
+      if ( record->event.pressed ) {
+
+        process_unicode( ( 0x0F3C|QK_UNICODE ), record );  /* Head */
+        tap_code( KC_SPACE );
+        process_unicode( ( 0x3064|QK_UNICODE ), record );  /* Hand */
+        tap_code( KC_SPACE );
+        process_unicode( ( 0x25D5|QK_UNICODE ), record );  /* Eye */
+        tap_code16( S( KC_MINS ) );                        /* Mouth */
+        process_unicode( ( 0x25D5|QK_UNICODE ), record );  /* Eye */
+        tap_code( KC_SPACE );
+        process_unicode( ( 0x0F3D|QK_UNICODE ), record );  /* Head */
+        process_unicode( ( 0x3064|QK_UNICODE ), record );  /* Hand */
```

Comment:
You could actually use `send_unicode_hex_string` for this, if unicode is enabled.

Reply:
When replacing my `process_unicode()` calls to `send_unicode_hex_string()`, should I only use `send_unicode_hex_string()` for the consecutive calls to `process_unicode()` to combine them, or should I convert the "standard" characters like space, minus, etc. to their unicode equivalents so that it can be all output in a single call to `send_unicode_hex_string()`?

Also, what should become of this block? 

https://github.com/d4mation/qmk_firmware/blob/6c0e7f2fb1cfc28efbbd9dd3b9955ad8dacc89bc/users/d4mation/macros.c#L85-L87

Should I leave that bit alone, or would `send_unicode_hex_string()` be able to handle that value?

Reply:
`send_unicode_hex_string()` can handle the entire thing.  Like space is 0020 in unicode. 

You can see where I've done this, here:
https://github.com/d4mation/qmk_firmware/blob/6c0e7f2fb1cfc28efbbd9dd3b9955ad8dacc89bc/users/drashna/process_records.c#L96-L114

This does require unicode to be enabled, but it makes things a lot simpler